lighthouse-core/test/gather/gather-runner-test.js

```diff
@@ -679,6 +681,43 @@ describe('GatherRunner', function() {
     });
   });
 
+  describe('#checkForSecurityIssue', () => {
+    it('succeeds when page is secure', () => {
+      const secureSecurityState = {
+        securityState: 'secure',
+      };
+      GatherRunner.assertNoSecurityIssues(secureSecurityState);
+    });
+
+    it('fails when page is insecure', () => {
+      const insecureSecurityState = {
+        explanations: [
+          {
+            description: 'reason 1.',
+            securityState: 'insecure',
+          },
+          {
+            description: 'blah.',
+            securityState: 'info',
+          },
+          {
+            description: 'reason 2.',
+            securityState: 'insecure',
+          },
+        ],
+        securityState: 'insecure',
+      };
+      try {
+        GatherRunner.assertNoSecurityIssues(insecureSecurityState);
+        assert.fail('expected INSECURE_DOCUMENT_REQUEST LHError');
+      } catch (err) {
+        assert.equal(err.message, 'INSECURE_DOCUMENT_REQUEST');
```

Comment:
maybe check `err.code` as well/instead, since that's what we use for the top level `runtimeError`

Reply:
Done. Some other tests were just testing `.message`, so I added checks there too. lmk if I should just remove the `.message` assertions.